cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -26,33 +27,37 @@ def __init__(self, base_url: str, api_key: str, order_email_id: int, env: str):
         self.env = env
         self.session = self._get_session()
 
-    def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
+    @handle_client_errors
+    def create_ticket(self, ticket: TicketCreate, attachments: list[Path] = None) -> TicketResponse:
         """Create a ticket."""
-        LOG.debug(ticket.model_dump_json())
-        try:
+        ticket_data = ticket.model_dump(exclude_none=True)
+
+        if attachments:
+            files = [
+                ("attachments[]", (attachment.name, open(attachment, "rb")))
+                for attachment in attachments
+            ]
+            response: Response = self.session.post(
+                url=self._url(EndPoints.TICKETS), data=ticket_data, files=files
+            )
+        else:
             response: Response = self.session.post(
-                url=self._url(EndPoints.TICKETS),
-                json=ticket.model_dump(exclude_none=True),
+                url=self._url(EndPoints.TICKETS), json=ticket_data
             )
-            response.raise_for_status()
-            return TicketResponse.model_validate(response.json())
-        except RequestException as error:
-            LOG.error(f"Could not create ticket: {error}")
-            raise FreshdeskAPIException(error) from error
-        except ValidationError as error:
-            LOG.error(f"Response from Freshdesk does not fit model: {TicketResponse}.\n{error}")
-            raise FreshdeskModelException(error) from error
+
+        response.raise_for_status()
+        return TicketResponse.model_validate(response.json())
 
     def _url(self, endpoint: str) -> str:
         """Get the full URL for the endpoint."""
         return f"{self.base_url}{endpoint}"
 
+    @handle_client_errors
     def _get_session(self) -> Session:
-        """Configures and sets a session to be used for requests."""
         session = Session()
-        self._configure_retries(session)
         session.auth = (self.api_key, "X")
-        session.headers.update({"Content-Type": "application/json"})
+        retries = Retry(total=5, backoff_factor=1, status_forcelist=[429, 500, 502, 503, 504])
```

Comment:
If you configure the retries here you can remove the `_configure_retries` method completely. However, I think the `_configure_retries` is clearer so I would keep it.

Reply:
👍